rust/protocol/src/storage/inmem.rs

```diff
@@ -1,38 +1,52 @@
 //
-// Copyright 2020 Signal Messenger, LLC.
+// Copyright 2020-2022 Signal Messenger, LLC.
 // SPDX-License-Identifier: AGPL-3.0-only
 //
 
+//! Implementations for stores defined in [super::traits].
+//!
+//! These implementations are purely in-memory, and therefore most likely useful for testing
+//! right now.
+//!
+//! *TODO: Define a serialization scheme for all of the in-memory stores so that they can be
+//! persisted to file or elsewhere!*
+
 use crate::{
-    IdentityKey, IdentityKeyPair, PreKeyRecord, ProtocolAddress, Result, SenderKeyRecord,
-    SessionRecord, SignalProtocolError, SignedPreKeyRecord,
+    address::ProtocolAddress,
+    error::{Result, SignalProtocolError},
+    sender_keys::SenderKeyRecord,
+    state::{PreKeyId, PreKeyRecord, SessionRecord, SignedPreKeyId, SignedPreKeyRecord},
+    storage::traits::{self, Context},
+    IdentityKey, IdentityKeyPair,
 };
 
-use crate::state::{PreKeyId, SignedPreKeyId};
-use crate::storage::traits;
-use crate::storage::Context;
-
 use async_trait::async_trait;
 use std::borrow::Cow;
 use std::collections::HashMap;
 use uuid::Uuid;
 
 #[derive(Clone)]
+/// Reference implementation of [traits::IdentityKeyStore].
```

Comment:
Nitpick: Doc comments before attributes looks a little tidier to me.

Reply:
IIRC, it's also what rustfmt would do.

Reply:
Done!